app/waf/src/controllers/application.controller.ts

```diff
@@ -220,6 +216,9 @@ export class ApplicationController extends BaseController {
       tenantId: tenantId,
     });
 
+    let mgmt = adc.management!;
```

Comment:
If you look at ADC controller, you will see that we also has a way to implicitly init ASG manager during controller object constructing. So that we will not have to explicitly create ASG manager object in every handler function.

Reply:
I actually intended to do that earlier, but it seems it didn't worked. Anyway, that's a different way to implement, I can try again later and will refactor that in another patch.

Reply:
now i remember, because asgService is not needed here and thus has already been removed from the constructor, so I now can not use the way which is used in AdcController now. So i will keep using this way.

Reply:
Ok. It is up to you